components/raftstore/src/store/peer.rs

```diff
@@ -704,6 +726,326 @@ pub enum UnsafeRecoveryState {
     Destroy(UnsafeRecoveryExecutePlanSyncer),
 }
 
+// #[RaftstoreCommon], copied from Peer
+/// Peer's trait
+pub trait RaftPeer<EK, ER>
+where
+    EK: KvEngine,
+    ER: RaftEngine,
+{
+    type Callback: ErrorCallback;
+
+    fn region(&self) -> &metapb::Region;
+    fn peer_id(&self) -> u64;
+    fn is_splitting(&self) -> bool;
+    fn is_merging(&self) -> bool;
+    fn is_leader(&self) -> bool;
+    fn term(&self) -> u64;
+    fn store_commit_index(&self) -> u64;
+    fn leader_id(&self) -> u64;
+    fn region_id(&self) -> u64;
+    fn tag(&self) -> &String;
+    fn txn_ext(&self) -> Arc<TxnExt>;
+    fn read_progress(&self) -> Arc<RegionReadProgress>;
+    fn peer(&self) -> &metapb::Peer;
+    fn bucket_meta(&self) -> Option<Arc<BucketMeta>>;
+    fn txn_extra_op(&self) -> Arc<AtomicCell<TxnExtraOp>>;
+    fn mut_pending_reads(&mut self) -> &mut ReadIndexQueue<Self::Callback>;
+    fn mut_bcast_wake_up_time(&mut self) -> &mut Option<TiInstant>;
+    fn set_should_wake_up(&mut self, should_wake_up: bool);
+    fn pending_remove(&self) -> bool;
+    fn has_force_leader(&self) -> bool;
+    fn mut_leader_lease(&mut self) -> &mut Lease;
+    fn store_applied_term(&self) -> u64;
+    fn store_applied_index(&self) -> u64;
+    fn has_pending_merge_state(&self) -> bool;
+    fn is_handling_snapshot(&self) -> bool;
+
+    fn pre_read_index(&self) -> Result<()> {
+        fail_point!(
+            "before_propose_readindex",
+            |s| if s.map_or(true, |s| s.parse().unwrap_or(true)) {
+                Ok(())
+            } else {
+                Err(box_err!(
+                    "{} can not read due to injected failure",
+                    self.tag()
+                ))
+            }
+        );
+
+        // See more in ready_to_handle_read().
+        if self.is_splitting() {
```

Comment:
You don't need to consider this ATM. This should be added when split/merge is supported. And they may not be the same as v1.